code/datums/job_selection/job_candidate.dm

```diff
@@ -0,0 +1,195 @@
+RESTRICT_TYPE(/datum/job_candidate)
+
+/// A datum representing all the information needed to perform roundstart job
+/// selection, without actually needing clients. This allows for the creation
+/// of fake candidates, allowing testing of the role selection system independent
+/// of the number of actual players.
+/datum/job_candidate
+	VAR_PRIVATE/datum/character_save/active_character
+	VAR_PRIVATE/datum/job_ban_holder/jbh
+	VAR_PRIVATE/list/be_special = list()
+	VAR_PRIVATE/list/restricted_roles = list()
+
+	VAR_PRIVATE/account_age_in_days
+	VAR_PRIVATE/admin_rights = 0
+	VAR_PRIVATE/assigned_job_title = null
+	VAR_PRIVATE/exp
+	VAR_PRIVATE/is_admin = FALSE
+	VAR_PRIVATE/is_guest_key = FALSE
+	VAR_PRIVATE/latejoin = FALSE
+	VAR_PRIVATE/real_ckey
+	VAR_PRIVATE/return_to_lobby = FALSE
+	VAR_PRIVATE/special_role
+
+/datum/job_candidate/New()
+	active_character = new()
```

Comment:
Make sure these are cleared on destroy if necessary

Reply:
done. i'm just nulling them out instead of qdeling them because they may be pointing to valid character saves we want around.